src/S3/S3Client.php

```diff
@@ -623,7 +623,9 @@ private function getLocationConstraintMiddleware()
         $region = $this->getRegion();
         return static function (callable $handler) use ($region) {
             return function (Command $command, $request = null) use ($handler, $region) {
-                if ($command->getName() === 'CreateBucket') {
+                if ($command->getName() === 'CreateBucket'
+                    && empty($command['CreateBucketConfiguration']['LocationConstraint'])
+                ) {
                     if ($region === 'us-east-1') {
```

Comment:
I think the behavior of removing the CreateBucketConfiguration structure should always be performed when the supplied value is `us-east-1`. Right now it's only removed in the provided value is empty. If the user supplies `us-east-1`, we should also remove that as that is the current behavior.